v3/impl.h

```diff
@@ -221,13 +228,36 @@ struct wide_integer<Bits, Signed>::_impl {
             r = -r;
         }
 
-        size_t count = r / std::numeric_limits<uint64_t>::max();
+        const long double div = r / max_int;
+        size_t count = max_sizet;
+
+        /// r / max_uint may not fit in size_t
+        if (div <= static_cast<long double>(max_sizet))
+            count = div;
+
         self = count;
-        self *= std::numeric_limits<uint64_t>::max();
+        self *= max_uint;
```

Comment:
Could you explain please: there is division by `max_int` in L231 but later `self` would be multiplied by `max_uint` in L239.
Why there is the difference?

MAYBE all `uint64_t` and `size_t` should be replaced with `int64_t`? Probably all type mismatches would be fixed without problems? Or not?
@apolukhin @myrrc

Reply:
> Could you explain please: there is division by `max_int` in L231 but later `self` would be multiplied by `max_uint` in L239.
> Why there is the difference?
> 
> MAYBE all `uint64_t` and `size_t` should be replaced with `int64_t`? Probably all type mismatches would be fixed without problems? Or not?
> @apolukhin @myrrc

There are some nasty cases as I've described before (rounding up the values of int64 and uint64 when converting them to double may lead to disability to convert them back to uint64/int64), so this question needs a more thorough review.